lib/screens/v2/widget_instance/dup_alert/serialize.ex

```diff
@@ -0,0 +1,199 @@
+      ["Blue"] -> :blue
+      _multiple_lines -> :yellow
+    end
+  end
+
+  defp get_line_text_builder(t) do
+    case DupAlert.get_affected_lines(t) do
+      [line_color] ->
+        fn _ -> [bold("#{line_color} Line")] end
+
+      [line_color1, line_color2] ->
+        fn and_or -> [bold("#{line_color1} Line"), and_or, bold("#{line_color2} Line")] end
+    end
+  end
+
+  defp partial_alert_free_text(t) do
+    build_line_text = get_line_text_builder(t)
+
+    affected_line_count = length(DupAlert.get_affected_lines(t))
+
+    case {affected_line_count, t.alert.effect, BaseAlert.location(t)} do
+      {1, :delay, _} ->
+        build_line_text.("and") ++ ["delays"]
+
+      {1, _, :inside} ->
+        ["No"] ++ build_line_text.("or") ++ ["trains"]
+
+      {1, _, boundary} when boundary in [:boundary_upstream, :boundary_downstream] ->
+        headsign = get_headsign(t)
+
+        ["No", bold(headsign), "trains"]
+        |> partial_headsign_special_cases()
+
+      {2, :delay, _} ->
+        ["Train delays"]
+
+      {2, _, _} ->
+        ["No train service"]
+    end
+  end
+
+  defp issue_free_text(t) do
+    build_line_text = get_line_text_builder(t)
+
+    case length(DupAlert.get_affected_lines(t)) do
+      1 ->
+        if BaseAlert.location(t) in [:boundary_upstream, :boundary_downstream] do
+          headsign = get_headsign(t)
+
+          ["No"] ++ build_line_text.("or") ++ ["trains to #{headsign}"]
+        else
+          ["No"] ++ build_line_text.("or") ++ ["trains", small(get_cause_string(t.alert.cause))]
+        end
+
+      2 ->
+        ["No"] ++ build_line_text.("or") ++ ["trains"]
+    end
+  end
+
+  defp remedy_free_text_line(t) do
+    icon = remedy_icon(t)
+    text = remedy_free_text(t)
+
+    %FreeTextLine{icon: icon, text: text}
+  end
+
+  defp remedy_icon(t) do
+    case t.alert.effect do
+      :shuttle -> :shuttle
+      _ -> nil
+    end
+  end
+
+  defp remedy_free_text(t) do
+    text =
+      case t.alert.effect do
+        :shuttle -> "Use shuttle bus"
+        _ -> "Seek alternate route"
+      end
+
+    [bold(text)]
+  end
+
+  defp get_headsign(t) do
+    headsign = BaseAlert.get_headsign_from_informed_entities(t)
+
+    if is_nil(headsign) do
+      raise("[DUP v2 no headsign] Could not determine headsign for DUP alert")
+    end
+
+    case headsign do
+      {:adj, hs} -> hs
+      hs -> hs
+    end
+  end
+
+  @alert_cause_mapping %{
+    accident: "an accident",
+    construction: "construction",
+    disabled_train: "a disabled train",
+    fire: "a fire",
+    holiday: "the holiday",
+    maintenance: "maintenance",
+    medical_emergency: "a medical emergency",
+    police_action: "police action",
+    power_problem: "a power issue",
+    signal_problem: "a signal problem",
+    snow: "snow conditions",
+    special_event: "a special event",
+    switch_problem: "a switch problem",
+    track_problem: "a track problem",
+    traffic: "traffic",
+    weather: "weather conditions"
+  }
```

Comment:
Copied and adapted from DupScreenData.Response; I think ReconstructedAlert did the same! 😅

I agree that it should probably live in one place instead of 3, lol. I'll find a spot for it.